packages/core/content-manager/server/tests/relations-dp.test.e2e.js

```diff
@@ -0,0 +1,260 @@
+'use strict';
+
+const { createTestBuilder } = require('../../../../../test/helpers/builder');
+const { createStrapiInstance } = require('../../../../../test/helpers/strapi');
+const { createAuthRequest } = require('../../../../../test/helpers/request');
+
+let strapi;
+let rq;
+const data = {
+  products: [],
+  shops: [],
+};
+
+const compo = {
+  displayName: 'compo',
+  attributes: {
+    name: {
+      type: 'string',
+    },
+    compoProducts: {
+      type: 'relation',
+      relation: 'manyToMany',
+      target: 'api::product.product',
+    },
+  },
+};
+
+const productModel = {
+  draftAndPublish: true,
+  attributes: {
+    name: {
+      type: 'string',
+    },
+  },
+  displayName: 'Product',
+  singularName: 'product',
+  pluralName: 'products',
+  description: '',
+  collectionName: '',
+};
+
+const shopModel = {
+  draftAndPublish: true,
+  attributes: {
+    name: {
+      type: 'string',
+    },
+    products: {
+      type: 'relation',
+      relation: 'manyToMany',
+      target: 'api::product.product',
+      targetAttribute: 'shops',
+      inversedBy: 'shops',
+    },
+    myCompo: {
+      type: 'component',
+      repeatable: false,
+      component: 'default.compo',
+    },
+  },
+  displayName: 'Shop',
+  singularName: 'shop',
+  pluralName: 'shops',
+};
+
+describe('Relations with Draft & Publish', () => {
+  const builder = createTestBuilder();
+
+  beforeAll(async () => {
+    await builder
+      .addContentTypes([productModel])
+      .addComponent(compo)
+      .addContentTypes([shopModel])
+      .build();
+
+    strapi = await createStrapiInstance();
+    rq = await createAuthRequest({ strapi });
+
+    const { body: createdProduct1 } = await rq({
```

Comment:
any reason not to use the utils to create the data here ?

Reply:
Like there : https://github.com/strapi/strapi/pull/14014/files#diff-db4c4860c0e75f128e57171dd85d026f21de601c2f92b24cf9538db56d07a62aR64 ?
- if yes : indeed I forgot
- if no because you think of the test helpers and the fixture utils : it's because I'm not comfortable using the fixture API, I better understand and I'm quicker the way I did. If that's a problem I can take time to better knex the helpers.

Reply:
I was more thinking about the fixtures utils indeed. They might need some documenting to make them easier to use I agree.

Reply:
You can skip it for now and just do the simpler createEntity function yes